src/assets/src/components/common.tsx

```diff
@@ -274,38 +271,45 @@ interface SingleInputFormShowRemainingProps extends StatelessSingleInputFormProp
 }
 
 export const SingleInputFormShowRemaining: React.FC<SingleInputFormShowRemainingProps> = (props) => {
-    const inputRef = createRef<HTMLTextAreaElement>();
-    useEffect(() => {
-        if (!props.autofocus) return;
-        inputRef.current!.focus();
-    }, []);
-    const submit = (e: React.FormEvent<HTMLFormElement>) => {
-        e.preventDefault();
-        props.onSubmit(props.value);
-        props.onChangeValue("");
+    const [remaining, setRemaining] = useState(props.maxLength - props.value.length as number)
+    const [isInvalid, setIsInvalid] = useState(false as boolean)
+    const buttonClass = `btn btn-${props.buttonType} remaining-controls` as string
+
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault()
+        props.onSubmit(props.value)
+        props.onChangeValue('')
     }
-    const buttonClass = "btn btn-" + props.buttonType + " remaining-controls";
-    const [remaining, setRemaining] = useState(props.maxLength - props.value.length);
-    const handleChange = (newValue:string) => {
+    const handleChange = (newValue: string) => {
         props.onChangeValue(newValue);
+        if (newValue.length <= props.maxLength) { setIsInvalid(false) } else { setIsInvalid(true) }
         setRemaining(props.maxLength - newValue.length)
     }
+
+    const charsRemaining = (remaining > 0) ? remaining : 0 as number
+    const charsOver = (remaining < 0) ? ` (${remaining * -1} over limit)` : '' as string
+    const feedbackText = `Remaining characters: ${charsRemaining}/${props.maxLength}` + charsOver as string
+
     return (
-        <form onSubmit={submit}>
-            <textarea onChange={(e) => handleChange(e.target.value)} value={props.value}
-                ref={inputRef} className="form-control" placeholder={props.placeholder}
-                disabled={props.disabled} id={props.id} rows={5}/>
-            <div>
-                <p className="remaining-controls-group">
-                    <span>{remaining}/{props.maxLength}</span>
-                    <button className={buttonClass} type="submit" disabled={props.disabled}>
-                        {props.children}
-                    </button>
-                </p>
-                
+        <Form onSubmit={handleSubmit}>
+            <Form.Group>
+                <Form.Control
+                    id={props.id}
+                    as='textarea'
+                    rows={5}
+                    value={props.value}
+                    placeholder={props.placeholder}
+                    onChange={(e) => handleChange(e.currentTarget.value)}
+                    disabled={props.disabled}
+                    isInvalid={!!isInvalid}
```

Comment:
It's already a boolean, so no need to coerce (`!!`)

Reply:
Ah, thanks, I think this may have been a vestige from when I was I initializing as undefined. Certainly doesn't look needed any more, haha

Reply:
Done.